changelog-tool/changelog.pony

```diff
@@ -79,25 +84,25 @@ class Release
     added = Section._empty(Added)
     changed = Section._empty(Changed)
 
+  fun ref add_entry(section_name: String, entry: String) ? =>
+    let section =
+      match section_name
+      | "fixed" => try fixed as Section else fixed = Section._empty(Fixed); fixed as Section end
+      | "added" => try added as Section else added = Section._empty(Added); added as Section end
+      | "changed" => try changed as Section else changed = Section._empty(Changed); changed as Section end
+      else error
+      end
+    section.entries.push("- " + entry)
+
   fun string(): String iso^ =>
-    if heading == _unreleased_heading then
```

Comment:
i don't think you meant to remove all of this.

Reply:
Well, if we don't remove this, it will print the _unrelease_ thing with an empty body, without the entries that we're adding. That's why I removed it 👍

Reply:
Doesn't this break the "unreleased" command?

Reply:
The "unreleased" release should always show the fixed, added, and changed sections, even if they are empty.

Reply:
I suspect that this is the cause of the CI failure

Reply:
Yeah!

```markdown
## [unreleased] - unreleased

### Fixed


### Added


### Changed
```

It's yielding `## [unreleased] - unreleased` and makes the test fail.

I'm struggling to reach a point where I can print the empty sections only from unlreleased versions.

Reply:
Turns out that unreleased sections were being parsed as `None`, so I added an extra check for each section saying that if it's unreleased and the section is `None`, it will use the `Section._empty(....)` for printing.

Reply:
Also, for the parse test that's failing, a changelog must now contain all sections in "unreleased" because it can no longer assume that the "unreleased" release is either empty or nonexistent. It may be best to create a new class, `Unreleased` that uses `Section` instead of `(Section | None)` as field types. This would be simpler than treating it as any other release at this point.

Reply:
@ordepdev would you want to include the changes above in this PR? Of not, I can do it after this PR is merged.

Reply:
I would love to but I think it's better to let you do it and take a look at your changes after!